v2/auth/cachedauth/sign_in.go

```diff
@@ -0,0 +1,61 @@
+package cachedauth
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/SKF/go-utility/v2/auth"
+)
+
+var lock sync.Mutex
```

Comment:
Unsure if we need performance here but maybe consider an RWMutex?

Reply:
I don't think it will be a very big improvment, but now we we allow concurrent getTokens()